iml-api/src/command.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::error::ImlApiError;
+use futures::TryFutureExt;
+use iml_orm::{
+    command::ChromaCoreCommand,
+    job::{get_jobs_by_cmd, ChromaCoreJob},
+    step::ChromaCoreStepresult,
+    tokio_diesel::AsyncRunQueryDsl as _,
+    DbPool,
+};
+use iml_wire_types::{Command, EndpointName as _, TestHostJob};
+use itertools::Itertools;
+
+pub(crate) async fn get_command(pool: &DbPool, id: i32) -> Result<Command, ImlApiError> {
+    let mut cmd: Vec<ChromaCoreCommand> = ChromaCoreCommand::by_id(id)
+        .get_results_async(&pool)
+        .map_err(ImlApiError::ImlDieselAsyncError)
+        .await?;
+
+    let cmd = cmd.pop().unwrap();
```

Comment:
```suggestion
         let cmd: Option<ChromaCoreCommand> = ChromaCoreCommand::by_id(id)
            .first_async(&pool)
            .await
            .optional()
            .ok_or_else(|| ImlApiError::NoneError);
```

This requires bringing `tokio_diesel::OptionalExtension as _` into scope and creating a new error variant to handle conversion of `None` into an `ImlApiError`.